src/IO/H5/StellarCollapseEos.hpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class h5::StellarCollapseEos
+
+#pragma once
+
+#include <hdf5.h>
+#include <string>
+#include <vector>
+
+#include "DataStructures/BoostMultiArray.hpp" // IWYU pragma: keep
+#include "IO/H5/Object.hpp"
+#include "IO/H5/OpenGroup.hpp"
+
+// IWYU pragma: no_include <boost/multi_array.hpp>
+
+namespace h5 {
+/*!
+ * \ingroup HDF5Group
+ * \brief Reads in tabulated equation of state file from
+ * <a href="https:://stellarcollapse.org"> stellarcollapse.org </a>
+ *
+ * Reads in H5 file containing data for tabulated equation of state.
+ * Contains functions to obtain thermodynamic quantities from the file,
+ * stored as either rank-1 or rank-3 datasets.
+ *
+ * It is assumed that the file is in the format of the SRO
+ * (Schneider, Roberts, Ott 2017) Equation of State files at
+ * <a href="https://stellarcollapse.org/SROEOS">
+ * stellarcollapse.org </a>
+ *
+ * The description of each dataset in the file can be found
+ * [here](https://bitbucket.org/andschn/sroeos/src/master/
+ * User_Guide/User_Guide.pdf?fileviewer=file-view-default)
+ *
+ */
+class StellarCollapseEos : public h5::Object {
+ public:
+  /// \cond HIDDEN_SYMBOLS
+  // The stellar collapse files do not have an extension when they are at the
+  // root level ('/') and we currently do not have any combined stellar
+  // collapse files.
+  static std::string extension() noexcept { return ""; }
+
+  StellarCollapseEos(bool exists, detail::OpenGroup&& group, hid_t location,
+                     const std::string& /*name*/) noexcept;
+
+  StellarCollapseEos(const StellarCollapseEos& /*rhs*/) = delete;
+  StellarCollapseEos& operator=(const StellarCollapseEos& /*rhs*/) = delete;
+  StellarCollapseEos(StellarCollapseEos&& /*rhs*/) noexcept = delete;
+  StellarCollapseEos& operator=(StellarCollapseEos&& /*rhs*/) noexcept = delete;
+
+  ~StellarCollapseEos() override = default;
+  /// \endcond HIDDEN_SYMBOLS
+
+  /*!
+   * \ingroup HDF5Group
+   * \fn template StellarCollapseEos::get_scalar_dataset(const std::string&
+   * dataset_name) const noexcept;
```

Comment:
What's this for?  The Doxygen docs say `\fn` isn't needed here and should not be used unless necessary.